Editor/PreviewSystem/ComputeContext/SingleObjectQueries.cs

```diff
@@ -37,18 +35,11 @@ public static GameObject GetAvatarRoot(
             GameObject candidate = null;
             foreach (var elem in context.ObservePath(obj.transform))
             {
-#if NDMF_VRCSDK3_AVATARS
-                if (context.GetComponent<VRCAvatarDescriptor>(elem.gameObject) != null)
-                {
-                    candidate = elem.gameObject;
-                    break;
-                }
-#else
                 if (context.GetComponent<Animator>(elem.gameObject) != null)
                 {
                     candidate = elem.gameObject;
+                    if (RuntimeUtil.IsAvatarRoot(elem)) break;
```

Comment:
This needs to observe the animator component (`context.GetComponent<Animator>` or similar) as well as any properties used to determine if the animator is used as an animator root.

It's also preferable to avoid observing the same object multiple times, as each observation registers a callback used to determine if something changed...

Reply:
It looks like essentially same as adding / removing VRCAvatarDescriptor in the middle of avatar hierarchy, so I need a strict (not a heuristic) definition of what is an avatar to continue working on this change.

Also, NDMF Avatar Root (as in #432 ) is not what I can actually start implementing without prior detailed design, because the change will certainly break [ResoniteImportHelper](https://github.com/KisaragiEffective/ResoniteImportHelper) and [Chillaxins](https://docs.hai-vr.dev/docs/products/chillaxins), and breaking non-VRChat support for a long time is not what I want to do.